Assets/HoloToolkit/Input/Scripts/Utilities/ControllerVisualizer.cs

```diff
@@ -51,51 +51,109 @@ public class ControllerVisualizer : MonoBehaviour
 
         private void Start()
         {
+            Application.onBeforeRender += Application_onBeforeRender;
+
             controllerDictionary = new Dictionary<uint, ControllerInfo>();
 
 #if UNITY_WSA
+            if (!Application.isEditor)
+            {
+                if (GLTFMaterial == null)
+                {
+                    if (LeftControllerOverride == null && RightControllerOverride == null)
+                    {
+                        Debug.Log("If using glTF, please specify a material on " + name + ". Otherwise, please specify controller overrides.");
+                    }
+                    else if (LeftControllerOverride == null || RightControllerOverride == null)
+                    {
+                        Debug.Log("Only one override is specified, and no material is specified for the glTF model. Please set the material or the " + ((LeftControllerOverride == null) ? "left" : "right") + " controller override on " + name + ".");
+                    }
+                }
+
 #if !UNITY_EDITOR
-            if (GLTFMaterial == null)
+                // Since the SpatialInteractionManager exists in the current CoreWindow, this call needs to run on the UI thread.
+                UnityEngine.WSA.Application.InvokeOnUIThread(() =>
+                {
+                    spatialInteractionManager = SpatialInteractionManager.GetForCurrentView();
```

Comment:
So do you know if there are any plans for Unity to expose more of these spatial APIs in the editor?

Reply:
As far as I'm aware, there's a plan to expose a GetNativeModel (or something) API, which will stop us needing to call the platform API. I'm also investigating using a native DLL to get/load the models, which should work in the editor as well.

Reply:
That would be amazing. Really, Unity should be footing the bill to implement some of this stuff though, to be honest.

Reply:
If the `InteractionManager` is just a wrapper for `SpatialInteractionManager` what's the advantage of subscribing to the detected and lost events here?

Reply:
InteractionManager doesn't provide a way to get the controller model. The only way to get it at the moment is through SpatialInteractionManager

Reply:
Couldn't we just make an extension to get the `source.controller` like we did with enabling and disabling the haptics?  That way everything can compile nicer in this class?

Reply:
Infact, we could add to the `HapticsExtensions` class and maybe rename it to be `InteractionSourceExtensions` or something along those lines.

Reply:
Hrm yeah, that's a good idea! I'll look into adding those extension methods.

Reply:
We should also probably move that class into `Input/Scripts/Utilities` as well while we're at it.